rules/event-reference.js

```diff
@@ -0,0 +1,110 @@
+const {
+  isAny, is
+} = require('bpmnlint-utils');
+
+
+/**
+ * A rule that verifies that event references are properly used.
+ *
+ * This implies:
+ *
+ *   * event references have a name
+ *   * event references are in use
+ *   * there exists only a single event reference
+ *     with a given name, per scope, per type
+ *
+ */
+module.exports = function() {
+
+  function check(node, reporter) {
+
+    if (!isAny(node, [
+      'bpmn:Error',
+      'bpmn:Escalation',
+      'bpmn:Message'
+    ])) {
+      return;
+    }
+
+    if (!node.name || node.name.trim() === '') {
+      reporter.report(node.id, 'Event reference is missing name');
+    }
+
+    const rootElement = getRootElement(node);
```

Comment:
You can then return `false` to stop further drill-down.

Reply:
> I'd rather implement this on `bpmn:Definitions` and search for globals; this ensures we don't rely on the hidden `$parent` property to discover the elements; it also will be computationally less expensive, as we have to compute `bpmnElements` only once, rather than `n` times for `n` global elements.

I _hope_ I understood you correctly. I am now traversing from `bpmn:Definitions` downwards, collecing events and event definitions on the way and do the validation. See [aad2302](https://github.com/bpmn-io/bpmnlint/pull/137/commits/aad2302512a4504ccf458245e93d8f1831917409)

Reply:
You _did_ understand me correctly. :slightly_smiling_face: